From 69dd2296d84426aa9ab7fe675770b6a590989be7 Mon Sep 17 00:00:00 2001
From: nxa15299 <Heinz.Wrobel@nxp.com>
Date: Fri, 12 Jan 2018 14:40:30 +0100
Subject: [PATCH] vnet: Fixed up cache flush to use kernel 4.9 API

Signed-off-by: nxa15299 <Heinz.Wrobel@nxp.com>
---
 ls/fpx.c  |  4 ++--
 s32/fpx.c | 11 +++--------
 2 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/drivers/net/vnet/ls/fpx.c b/drivers/net/vnet/ls/fpx.c
index fdc501e..6ee6a1d 100644
--- a/drivers/net/vnet/ls/fpx.c
+++ b/drivers/net/vnet/ls/fpx.c
@@ -29,7 +29,7 @@
 
 #define DRIVER_NAME	"fpx"
 
-extern void __dma_flush_range(const void *, const void *);
+extern void __dma_flush_area(const void *, size_t);
 
 static netdev_tx_t
 fpx_enet_start_xmit(struct sk_buff *skb, struct net_device *ndev)
@@ -54,7 +54,7 @@ fpx_enet_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 		void* end = start + (u32)skb->len + sizeof(u32);
 
 		*(u32*)(skb->data - sizeof(u32)) = (u32)skb->len;
-		__dma_flush_range((const void*)start, (const void*)end);
+		__dma_flush_area(start, skb->len + sizeof(u32));
 
 		*fep->qdma_regs &= ~1;
 		status = *(fep->qdma_regs + 1) & 0x92;
diff --git a/drivers/net/vnet/s32/fpx.c b/drivers/net/vnet/s32/fpx.c
index f7bf148..022fe0c 100644
--- a/drivers/net/vnet/s32/fpx.c
+++ b/drivers/net/vnet/s32/fpx.c
@@ -52,11 +52,9 @@ struct s32v_outbound_region {
 
 extern int s32v_pcie_setup_outbound(void *data);
 
-extern void __dma_flush_range(const void *, const void *);
+extern void __dma_flush_area(const void *, size_t);
 extern void __inval_cache_range(const void *, const void *);
 
-
-
 static netdev_tx_t
 fpx_enet_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 {
@@ -109,7 +107,7 @@ fpx_enet_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 			*(int*)(start) = skb->len;
 
 			/* flush cache */
-			__dma_flush_range((const void*)start, (const void*)end);
+			__dma_flush_area(start, skb->len + sizeof(int));
 
 			ndev->stats.tx_packets++;
 			ndev->stats.tx_bytes += skb->len;
@@ -137,10 +135,7 @@ fpx_enet_start_xmit(struct sk_buff *skb, struct net_device *ndev)
 
 			/* do TX */
 			/* Program DMA regs for LL mode */
-			start = (void*)&fep->d_tx[0];
-			end = (void*)&fep->d_tx[3];
-
-			__dma_flush_range((const void*)start, (const void*)end);
+			__dma_flush_area(&fep->d_tx[0], 3 * sizeof(fep->d_tx[0]));
 			fep->transmiter_status = STS_TX_INPROGRESS;
 			dw_start_dma_llw(fep->pp, virt_to_phys(fep->d_tx));
 		}
-- 
1.8.3.1

