From d7cb0f4094e345d178424f55cf8c11ee5fc48182 Mon Sep 17 00:00:00 2001
From: r39252 <Heinz.Wrobel@freescale.com>
Date: Wed, 10 Jan 2018 13:35:38 +0100
Subject: [PATCH] vnet: Added support for S32V with encryption enabled

---
 drivers/net/vnet/ls/fpx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/vnet/ls/fpx.c b/drivers/net/vnetls/fpx.c
index edddb35..fdc501e 100644
--- a/drivers/net/vnet/ls/fpx.c
+++ b/drivers/net/vnet/ls/fpx.c
@@ -439,6 +439,9 @@ static void fpx_remove(struct pci_dev *pdev)
 
 static const struct pci_device_id fpx_ids[] = {
 {
+	PCI_DEVICE(0x1957, 0x4000)
+},
+{
 	PCI_DEVICE(0x1957, 0x4001)
 },
 {0, }
-- 
1.8.3.1

