From 34441f1ab3ace726b6d732a4b90dea57e2d0e9ee Mon Sep 17 00:00:00 2001
From: BJ DevOps Team <bjdevops@NXP1.onmicrosoft.com>
Date: Mon, 4 Nov 2019 10:45:09 +0800
Subject: [PATCH] lpc_intrin_vsx.c: remove altvex

---
 src/libFLAC/lpc_intrin_vsx.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/libFLAC/lpc_intrin_vsx.c b/src/libFLAC/lpc_intrin_vsx.c
index 48c8218..1ca5c26 100644
--- a/src/libFLAC/lpc_intrin_vsx.c
+++ b/src/libFLAC/lpc_intrin_vsx.c
@@ -43,7 +43,6 @@
 #include "FLAC/assert.h"
 #include "FLAC/format.h"
 
-#include <altivec.h>
 
 #ifdef FLAC__HAS_TARGET_POWER8
 __attribute__((target("cpu=power8")))
-- 
2.17.1

