From 2a9924c81829a5f50805c6913b208286136fcb20 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Thu, 20 Dec 2018 13:14:56 +0200
Subject: [PATCH] rcw: Enabled A-008851 workaround for Gen3 RCW on LS1046ARDB

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Signed-off-by: Dan Nica <dan.nica@nxp.com>
---
 ls1046ardb/pex_gen3_link.rcw | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/ls1046ardb/pex_gen3_link.rcw b/ls1046ardb/pex_gen3_link.rcw
index 27e7459..e69d051 100644
--- a/ls1046ardb/pex_gen3_link.rcw
+++ b/ls1046ardb/pex_gen3_link.rcw
@@ -12,10 +12,25 @@
  *
  * ALTCBAR Register (SCFG_ALTCBAR): 0x0157_0158 big endian
  * PBL is also big endian block on LS1046A
+ *
+ * Errata A-008851 and A-010477
  */
 
 .pbi
 write 0x570158, 0x00000300
+awrite 0x4008bc,0x01000000
+awrite 0x5008bc,0x01000000
+awrite 0x6008bc,0x01000000
+awrite 0x400154,0x47474747
+awrite 0x400158,0x47474747
+awrite 0x500154,0x47474747
+awrite 0x500158,0x47474747
+awrite 0x600154,0x47474747
+awrite 0x600158,0x47474747
+awrite 0x4008bc,0x00000000
+awrite 0x5008bc,0x00000000
+awrite 0x6008bc,0x00000000
+
 awrite 0x400890, 0x01048000
 awrite 0x500890, 0x01048000
 awrite 0x600890, 0x01048000
-- 
2.7.4

