From 1d86003247bf9af8619cc3b0cc57746211f49035 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Wed, 25 Oct 2017 10:10:42 +0200
Subject: [PATCH] rcw: Enabled proper RCW swap for functional QSPI boot

---
 ls1046ardb/ls1046a.rcwi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ls1046ardb/ls1046a.rcwi b/ls1046ardb/ls1046a.rcwi
index a1dc700..258b439 100644
--- a/ls1046ardb/ls1046a.rcwi
+++ b/ls1046ardb/ls1046a.rcwi
@@ -1,6 +1,8 @@
 %size=512
 %sysaddr=ee0100
 %pbladdr=610000
+%littleendian64b=1
+%dont64bswapcrc=1
 
 SYS_PLL_CFG[0:1]
 SYS_PLL_RAT[2:6]
-- 
1.7.12.4

