From 129dbc045cbc6114b305f319650d30c775486744 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Wed, 19 Dec 2018 17:04:28 +0200
Subject: [PATCH] rcw: Added meta-bluebox related boards to the Makefile

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Signed-off-by: Dan Nica <dan.nica@nxp.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 9f0587e..7e51af6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,6 @@
 DESTDIR = .
 BOARDS = b4420qds b4860qds \
+	 ls2080abluebox ls2084abluebox ls2084abbmini \
 	 ls1012ardb ls1012a2g5rdb ls1012afrdm ls1012aqds \
 	 ls1021aqds ls1021atwr ls1043aqds ls1043ardb ls1046ardb ls1046aqds \
 	 ls1088ardb ls1088aqds\
-- 
2.7.4

