From 764acb0f43979e97553e4915ea0e5106bd960251 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Wed, 17 May 2017 11:49:55 +0200
Subject: [PATCH] rcw: Added meta-bluebox related boards to the Makefile

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index fb55c8b..ee87bb3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,7 @@
 DESTDIR = .
 BOARDS = b4420qds b4860qds \
+	 ls1012ardb \
+	 ls2080abluebox ls2084abluebox ls2084abbmini \
 	 ls1021aqds ls1021atwr ls1043aqds ls1043ardb ls1046ardb ls1046aqds \
 	 p2041rdb p3041ds p4080ds p5020ds p5040ds \
 	 t1024qds t1023rdb t1024rdb t1040rdb t1042rdb t1042rdb_pi t1040qds \
-- 
1.7.12.4

