From 05794433c54921bcc9c86a081d1214e8852a72a7 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Mon, 3 Apr 2017 19:07:42 +0200
Subject: [PATCH] rcw: Broke awrite for older pbi format in prior patch

---
 rcw.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rcw.py b/rcw.py
index 9b9f785..dabb959 100755
--- a/rcw.py
+++ b/rcw.py
@@ -328,7 +328,7 @@ def build_pbi(lines):
             if pbiformat == 2:
                 v1 = struct.pack(endianess + 'L', 0x80000000 + (opsizebytes << 26) + p1)
             else:
-                v1 = struct.pack( + 'L', 0x89000000 + p1)
+                v1 = struct.pack(endianess + 'L', 0x89000000 + p1)
             v2 = struct.pack(endianess + 'L', p2)
             subsection += v1
             subsection += v2
-- 
1.7.12.4

