From 2e650bb44838a46b00603e0f6b0a5d3aaa493b33 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@freescale.com>
Date: Wed, 29 Mar 2017 22:08:30 +0200
Subject: [PATCH 2/2] Corrected blockcopy command for fetching SDHC U-Boot

---
 .../rcw_ffffpphh_0x2a_0x41_SDHC_1800_700_1867_1600.rcw                  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ls2084abbmini/FFFFPPHH_0x2a_0x41/rcw_ffffpphh_0x2a_0x41_SDHC_1800_700_1867_1600.rcw b/ls2084abbmini/FFFFPPHH_0x2a_0x41/rcw_ffffpphh_0x2a_0x41_SDHC_1800_700_1867_1600.rcw
index 02b3d92..60471cc 100644
--- a/ls2084abbmini/FFFFPPHH_0x2a_0x41/rcw_ffffpphh_0x2a_0x41_SDHC_1800_700_1867_1600.rcw
+++ b/ls2084abbmini/FFFFPPHH_0x2a_0x41/rcw_ffffpphh_0x2a_0x41_SDHC_1800_700_1867_1600.rcw
@@ -36,7 +36,7 @@ SRDS_PRTCL_S2=65
 /* PBI_LENGTH=0 */
 #include <markscratchrw1.rcw>
 .pbi
-blockcopy 0x00100000,0x1800a000,0x00015000
+blockcopy 0x40,0x00100000,0x1800a000,0x00015000
 .end
 #include <bootlocptr_sdhc.rcw>
 #include <a009102.rcw>
-- 
2.0.4

