diff --git a/t4bluebox/Makefile b/t4bluebox/Makefile
index e69de29..f77e46b 100644
--- a/t4bluebox/Makefile
+++ b/t4bluebox/Makefile
@@ -0,0 +1 @@
+include ../Makefile.inc
diff --git a/t4bluebox/README b/t4bluebox/README
index e69de29..5534067 100644
--- a/t4bluebox/README
+++ b/t4bluebox/README
@@ -0,0 +1,16 @@
+The RCW directories names for the T4240RDB board conform to the following
+naming convention:
+
+abcdefg_h_i_j_k (default SSFFPPH_27_55_1_9) means:
+a   = 'S'  indicates FM1 @ DTSEC1 DTSEC2 DTSEC3 DTSEC4 ports are supported
+b   = 'S'  indicates FM2 @ DTSEC1 DTSEC2 DTSEC3 DTSEC4 ports are supported
+c   = 'F'  indicates FM1 @ TGEC1 TGEC2 SFP+ fiber ports are supported
+d   = 'F'  indicates FM2 @ TGEC1 TGEC2 SFP+ fiber ports are supported
+e   = 'P'  indicates PCIe x8 in the PCI slot
+f   = 'P'  indicates PCIe x4 in the PCI slot
+g   = 'H'  indicates SATA port is supported
+h   = '27' indicates SerDes1 protocol 0x1B
+i   = '55' indicates SerDes2 protocol 0x37
+j   = '1'  indicates SerDes3 protocol 0x1
+k   = '9'  indicates SerDes4 protocol 0x9
+
diff --git a/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1666MHz.rcw b/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1666MHz.rcw
index e69de29..ca8df43 100644
--- a/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1666MHz.rcw
+++ b/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1666MHz.rcw
@@ -0,0 +1,62 @@
+/*
+ * T4240RDB RCW for SerDes[1:4] Protocol 27, 55, 1, 9
+ *
+ * 48G configuration -- 8 SGMII + 4 XFI
+ *
+ * Frequencies:
+ *
+ * Sys Clock -- 66.67 Mhz
+ * SDREFCLK1_FSEL: 100 MHz
+ * SDREFCLK2_FSEL: 156.25 MHz
+ * SDREFCLK3_FSEL: 100 MHz
+ * SDREFCLK3_FSEL: 100 MHz
+ *
+ * Core		-- 1666 MHz
+ * Platform	-- 733 MHz
+ * DDR		-- 933 MHz
+ * FMAN1	-- 733 MHz
+ * FMAN2	-- 733 MHz
+ * PME		-- 533MHz
+ *
+ * PBI source is I2C.
+ */
+
+#include <../t4240qds/t4240.rcwi>
+
+SYS_PLL_RAT=11
+MEM_PLL_RAT=7
+CGA_PLL1_RAT=25
+CGA_PLL2_RAT=24
+CGA_PLL3_RAT=16
+CGB_PLL1_RAT=25
+CGB_PLL2_RAT=22
+SRDS_PRTCL_S1=27
+SRDS_PRTCL_S2=55
+SRDS_PRTCL_S3=1
+SRDS_PRTCL_S4=9
+SRDS_PLL_PD_S1=1
+SRDS_PLL_PD_S3=1
+SRDS_DIV_PEX_S3=2
+SRDS_DIV_AURORA_S4=1
+SRDS_DIV_PEX_S4=2
+PBI_SRC=0
+BOOT_LOC=24
+IFC_MODE=32
+HWA_CGA_M1_CLK_SEL=7
+HWA_CGB_M1_CLK_SEL=5
+DRAM_LAT=1
+GP_INFO=3992977646
+UART_BASE=5
+IRQ_BASE=511
+EC1=2
+EC2=2
+HWA_CGB_M2_CLK_SEL=5
+
+// IFC bus speed work-around
+.pbi
+write 0x1241c0, 0xf03f3f3f
+write 0x1241c4, 0xff003f3f
+write 0x124010, 0x00000101
+write 0x124130, 0x0000000c
+.end
+
diff --git a/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1800MHz.rcw b/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1800MHz.rcw
index e69de29..5650049 100644
--- a/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1800MHz.rcw
+++ b/t4bluebox/SSFFPPH_27_55_1_9/rcw_27_55_1_9_1800MHz.rcw
@@ -0,0 +1,62 @@
+/*
+ * T4240RDB RCW for SerDes[1:4] Protocol 27, 55, 1, 9
+ *
+ * 48G configuration -- 8 SGMII + 4 XFI
+ *
+ * Frequencies:
+ *
+ * SDREFCLK1_FSEL: 100 MHz
+ * SDREFCLK2_FSEL: 156.25 MHz
+ * SDREFCLK3_FSEL: 100 MHz
+ * SDREFCLK3_FSEL: 100 MHz
+ *
+ * Sys Clock	-- 66.67 Mhz
+ * Core		-- 1800 MHz
+ * Platform	-- 733 MHz
+ * DDR		-- 933 MHz
+ * FMAN1	-- 733 MHz
+ * FMAN2	-- 733 MHz
+ * PME		-- 533 MHz
+ *
+ * PBI source is I2C.
+ */
+
+#include <../t4240qds/t4240.rcwi>
+
+SYS_PLL_RAT=11
+MEM_PLL_RAT=7
+CGA_PLL1_RAT=27
+CGA_PLL2_RAT=24
+CGA_PLL3_RAT=16
+CGB_PLL1_RAT=27
+CGB_PLL2_RAT=22
+SRDS_PRTCL_S1=27
+SRDS_PRTCL_S2=55
+SRDS_PRTCL_S3=1
+SRDS_PRTCL_S4=9
+SRDS_PLL_PD_S1=1
+SRDS_PLL_PD_S3=1
+SRDS_DIV_PEX_S3=2
+SRDS_DIV_AURORA_S4=1
+SRDS_DIV_PEX_S4=2
+PBI_SRC=0
+BOOT_LOC=24
+IFC_MODE=32
+HWA_CGA_M1_CLK_SEL=7
+HWA_CGB_M1_CLK_SEL=5
+DRAM_LAT=1
+GP_INFO=3992977646
+UART_BASE=5
+IRQ_BASE=511
+EC1=2
+EC2=2
+HWA_CGB_M2_CLK_SEL=5
+
+// IFC bus speed work-around
+.pbi
+write 0x1241c0, 0xf03f3f3f
+write 0x1241c4, 0xff003f3f
+write 0x124010, 0x00000101
+write 0x124130, 0x0000000c
+.end
+
