From 49dd0f6eda9d6d4075080caf5877f75f59d90600 Mon Sep 17 00:00:00 2001
From: Stefan-Gabriel Mirea <stefan-gabriel.mirea@nxp.com>
Date: Tue, 6 Mar 2018 11:50:16 +0200
Subject: [PATCH] opendds: Make Version.h symlink path relative

Signed-off-by: Stefan-Gabriel Mirea <stefan-gabriel.mirea@nxp.com>
---
 dds/DdsDcps.mpc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dds/DdsDcps.mpc b/dds/DdsDcps.mpc
index 138cf9c..58d00e1 100644
--- a/dds/DdsDcps.mpc
+++ b/dds/DdsDcps.mpc
@@ -87,7 +87,7 @@ project(OpenDDS_Dcps): core, coverage_optional, \
   verbatim(gnuace, postinstall) {
 "	echo export DDS_ROOT=$(DESTDIR)$(INSTALL_PREFIX)/share/dds> $(DESTDIR)$(INSTALL_PREFIX)/share/dds/dds-devel.sh"
 "	@$(MKDIR) $(DESTDIR)$(INSTALL_PREFIX)/share/dds/dds"
-"	ln -sf $(DESTDIR)$(INSTALL_PREFIX)/include/dds/Version.h $(DESTDIR)$(INSTALL_PREFIX)/share/dds/dds"
+"	ln -sf ../../../include/dds/Version.h $(DESTDIR)$(INSTALL_PREFIX)/share/dds/dds"
   }
 
 }
-- 
2.16.2

