From 90beebd4bd5f91ecbf004b12c7a2255348b07ecd Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Fri, 11 Nov 2022 15:15:28 +0200
Subject: [PATCH] qemu-user: strace didn't print faccessat() correctly

Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Signed-off-by: Ionut Vicovan <Ionut.Vicovan@nxp.com>
---
 linux-user/strace.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/linux-user/strace.c b/linux-user/strace.c
index b8560b511..a40ed81e1 100644
--- a/linux-user/strace.c
+++ b/linux-user/strace.c
@@ -1944,7 +1944,8 @@ print_faccessat(void *cpu_env, const struct syscallname *name,
     print_at_dirfd(arg0, 0);
     print_string(arg1, 0);
     print_flags(access_flags, arg2, 0);
-    print_flags(at_file_flags, arg3, 1);
+    /* Kernel does not receive arg3! We pretend! */
+    qemu_log("?");
     print_syscall_epilogue(name);
 }
 #endif
-- 
2.25.1

