From 89bcde95697e45f0355ee1896976dacb786db6d7 Mon Sep 17 00:00:00 2001
From: Vicovan Ionut-Valentin <Ionut.Vicovan@nxp.com>
Date: Thu, 7 Dec 2017 18:11:39 +0200
Subject: [PATCH] kvaser: fix parallel build

Signed-off-by: Vicovan Ionut-Valentin <Ionut.Vicovan@nxp.com>
Signed-off-by: Stefan-Gabriel Mirea <stefan-gabriel.mirea@nxp.com>
---
 Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index ba52682..f048296 100644
--- a/Makefile
+++ b/Makefile
@@ -120,31 +120,31 @@ print_versions_start:
 canlib:
 	$(MAKE) -C canlib examples
 
-linlib:
+linlib: canlib
 	$(MAKE) -C linlib
 
 common:
 	@cd ./common; $(MAKE) kv_module
 
-pcican:
+pcican: common
 	@cd ./pcican; $(MAKE) kv_module
 
-pcican2:
+pcican2: common
 	@cd ./pcican2; $(MAKE) kv_module
 
-usbcanII:
+usbcanII: common
 	@cd ./usbcanII; $(MAKE) kv_module
 
-leaf:
+leaf: common
 	@cd ./leaf; $(MAKE) kv_module
 
-mhydra:
+mhydra: common
 	@cd ./mhydra; $(MAKE) kv_module
 
-virtualcan:
+virtualcan: common
 	@cd ./virtualcan; $(MAKE) kv_module
 
-pciefd:
+pciefd: common
 	@cd ./pciefd; $(MAKE) kv_module
 
 install:
-- 
2.21.0

