# Copyright 2018 NXP
diff --git a/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_s32v234.c b/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_s32v234.c
index 74836f4..e16c81e 100644
--- a/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_s32v234.c
+++ b/hal/os/linux/kernel/platform/freescale/gc_hal_kernel_platform_s32v234.c
@@ -475,8 +475,8 @@ gckPLATFORM_AdjustParam(
 
     Args->gpu3DMinClock = initgpu3DMinClock;
 
-  if(Args->physSize == 0)
-    Args->physSize = 0x40000000;
+    if(Args->physSize == 0)
+       Args->physSize = 0x40000000;
 
     return gcvSTATUS_OK;
 }
-- 
1.9.1

